src/IndexService/IndexUpdateService.php

```diff
@@ -182,16 +182,18 @@ public function resetIdsUpdateIndex(array $idList, string $triggerInfo, array $t
      */
     protected function resetIds(array $idList, string $triggerInfo, bool $onlyResetUpdateIndex = false, array $tenantNameList = null): void
     {
-        if (count($idList) <= 0) {
+        if (count($idList) === 0) {
             return;
         }
 
-        if ($tenantNameList == null) {
-            $tenantNameList = $this->indexService->getTenants();
-        } elseif (count($tenantNameList) <= 0) {
+        if(is_countable($tenantNameList)  && count($tenantNameList) === 0) {
             return;
         }
 
+        if ($tenantNameList === null) {
+            $tenantNameList = $this->indexService->getTenants();
```

Comment:
i'd swap the order of these if block, because `$this->indexService->getTenants();` could be also `0` and likely also `return void` as this
https://github.com/pimcore/ecommerce-framework-bundle/blob/e6ca0e41fc37f22ffafec47660ea3faca975d66b/src/IndexService/IndexUpdateService.php#L189-L191

Reply:
I get your insight, the only problem is, I think then we change the behavior right? Because now the code just reflects the same as before, just structured a little different? Because the `getStoreTableList` then returns all stored table lists?
But like I said I am not so sure about the initial intention here :/

Reply:
mhhh, nice point, then my suggestion is not right. 
Let's just change `is_countable` to `is_array`, as it is impossible that is something else than array beside `null` which is not countable